libbeat/common/jsontransform/jsonhelper.go

```diff
@@ -77,6 +77,10 @@ func WriteJSONKeys(event *beat.Event, keys map[string]interface{}, expandKeys, o
 			event.Timestamp = ts
 
 		case "@metadata":
+			if event.Meta == nil {
+				event.Meta = common.MapStr{}
+			}
+
 			switch m := v.(type) {
 			case map[string]string:
```

Comment:
Is this needed in all cases or only in the `map[string]string`  case? I wonder if this should be left as `nil` in other cases.

```suggestion
			switch m := v.(type) {
			case map[string]string:
				if event.Meta == nil && len(m) > 0 {
					event.Meta = common.MapStr{}
				}
```

Reply:
Actually, my real case hit the `map[string]interface{}` case.

I modified the code only to update `event.Meta` in these two cases.

ref: https://github.com/elastic/beats/pull/24741/commits/c95f2c66303d0396c6a7ed28258736146d228437

Reply:
Thanks!